src/data/fintype/order.lean

```diff
@@ -0,0 +1,50 @@
+/-
+Copyright (c) 2021 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+
+import order.lattice
+import data.fintype.basic
+
+/-!
+# Order structures on finite sets
+
+This file provides instances showing that a `semilattice_sup_bot` on a `fintype` gives a
+`complete_lattice`, and that a `linear_order` on a nonempty `fintype` gives a
+`complete_linear_order`. The latter applies in particular to `fin (n+1)`. The instances are
+noncomputable because the definitions of `Sup` and `Inf` require passing from `set` to `finset`.
+-/
+
+open_locale classical
+
+variables {α : Type*}
+
+@[priority 100] noncomputable instance fintype.has_Sup [fintype α] [semilattice_sup_bot α] :
```

Comment:
What is it that makes this noncomputable?

Reply:
The same reason as in the module doc, it's the `to_finset` part.

Reply:
Isn't `to_finset` computable though?

Reply:
It relies on `subtype.fintype`, which needs a `decidable_pred` instance for the set in question. 

Mario says in this post that noncomputability is fine here. https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/fin.20n.20has_Inf.20and.20has_Sup/near/206633448

Reply:
(In other words, the real answer to your question is received wisdom)

Reply:
Capturing as much as that wisdom as possible in the module doc would be great. I see the point though, you'd need `decidable_pred (\mem s)` for all `s`, which is only possible with the axiom of choice

Reply:
Thanks - updated the docstring.

Reply:
For the record, `to_finset` is noncomputable, as in https://github.com/leanprover-community/mathlib/blob/aa1fa0b2ba0739deaec3fe65f42640998120bc80/src/data/set/finite.lean#L36.

Reply:
@b-mehta, that's `set.finite.to_finset` not `set.to_finset`

Reply:
Good point, my mistake